File: internal/models/workflow.go
package models

import (
	"time"
)

type WorkflowGroup struct {
	ID          uint       `json:"id" gorm:"primaryKey"`
	Name        string     `json:"name" gorm:"not null;size:100"`
	Description string     `json:"description,omitempty"`
	Status      string     `json:"status" gorm:"default:'active';size:20"` // active, archived
	CreatedBy   uint       `json:"created_by" gorm:"not null"`
	CreatedAt   time.Time  `json:"created_at" gorm:"autoCreateTime"`
	EndedAt     *time.Time `json:"ended_at,omitempty"`

	// 关联关系
	Creator   *User      `json:"creator,omitempty" gorm:"foreignKey:CreatedBy"`
	Materials []Material `json:"materials,omitempty" gorm:"foreignKey:WorkflowID"`
}

type WorkflowMember struct {
	ID         uint      `json:"id" gorm:"primaryKey"`
	WorkflowID uint      `json:"workflow_id" gorm:"not null"`
	UserID     uint      `json:"user_id" gorm:"not null"`
	Role       string    `json:"role" gorm:"not null;size:20"` // photographer, reviewer, admin
	JoinedAt   time.Time `json:"joined_at" gorm:"autoCreateTime"`

	// 关联关系
	Workflow *WorkflowGroup `json:"workflow,omitempty" gorm:"foreignKey:WorkflowID"`
	User     *User          `json:"user,omitempty" gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE;"`
}

File: MediaCloudDesktop/Models/Workflow.cs
using System.Text.Json.Serialization;

namespace MediaCloudDesktop.Models
{
    public class Workflow
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [JsonPropertyName("description")]
        public string Description { get; set; } = string.Empty;

        [JsonPropertyName("type")]
        public string Type { get; set; } = "custom";

        [JsonPropertyName("color")]
        public string Color { get; set; } = "#409EFF";

        [JsonPropertyName("is_active")]
        public bool IsActive { get; set; } = true;

        [JsonPropertyName("config")]
        public string Config { get; set; } = string.Empty;

        [JsonPropertyName("status")]
        public string Status { get; set; } = "active";

        [JsonPropertyName("created_by")]
        public int CreatedBy { get; set; }

        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; }

        [JsonPropertyName("creator")]
        public User? Creator { get; set; }

        [JsonPropertyName("members")]
        public List<WorkflowMember>? Members { get; set; }
    }

    public class WorkflowMember
    {
        [JsonPropertyName("user_id")]
        public int UserId { get; set; }

        [JsonPropertyName("role")]
        public string Role { get; set; } = "member";

        [JsonPropertyName("user")]
        public User? User { get; set; }
    }
}
